nixos/modules/services/networking/go-camo.nix

```diff
@@ -0,0 +1,62 @@
+{ lib, pkgs, config, ... }:
+with lib;
+
+let
+  cfg = config.services.go-camo;
+in
+{
+  options.services.go-camo = {
+    enable = mkEnableOption "go-camo service";
+    addressPort = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = "Address:Port to bind to for HTTP (default: 0.0.0.0:8080)";
+      apply = v: optionalString (v != null) "--listen=${v}";
+    };
+    sslListen = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = "Address:Port to bind to for HTTPS";
+      apply = v: optionalString (v != null) "--ssl-listen=${v}";
+    };
+    sslKey = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = "Path to TLS private key";
+      apply = v: optionalString (v != null) "--ssl-key=${v}";
+    };
+    sslCert = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = "Path to TLS certificate";
+      apply = v: optionalString (v != null) "--ssl-cert=${v}";
+    };
+    key = mkOption {
```

Comment:
Is this a secret?

Reply:
Correct. It's also accepted as `GOCAMO_HMAC` environment variable. I'll move it there instead of the argument.
Did you want to change something more about it?

Reply:
There should be a way to provide it via a file

Reply:
I'm struggling with this one. I can easily provide the key from the string or file this way:
```
mkMerge [
  ...
    EnvironmentFile = "/etc/go-camo.key";
  ...
  (mkIf (cfg.key != null) { environment.etc."go-camo.key".text = ("GOCAMO_HMAC=" + cfg.key); })
  (mkIf (cfg.keyFile != null) { environment.etc."go-camo.key".source = cfg.keyFile); })
]
```
But I don't know how to nicely prepend the "GOCAMO_HMAC=" to the file source. I can't do that in the `ExecStartPre` either since the `EnvironmentFile` is loaded before that's executed.
I could force it through another one-shot service that go-camo depends on, but... that sounds stupid.
Any suggestions?

Reply:
If you want full bonus points you could use [LoadCredential](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#LoadCredential=ID:PATH), the safe alternative to `EnvironmentFile`. Something *like* this:

```
systemd.services.go-camo = {
  script = ''
    env GOCAME_HMAC=''${CREDENTIALS_DIRECTORY}/key.file ${pkgs.go-camo}/bin/go-camo --options --here
  '';
  serviceConfig = {
    LoadCredential = "key.file:${cfg.keyFile}";
  };
};
```

I probably shouldn't review PRs when I'm this tired... @SuperSandro2000 - the above makes sense, right?